tests/integration/olp-cpp-sdk-core/OlpClientDefaultAsyncHttp.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gtest/gtest.h>
+
+
+#include <future>
+
+#include <olp/core/client/OlpClient.h>
+#include <olp/core/client/OlpClientFactory.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+
+#include <olp/core/http/Network.h>
+
+class OlpClientDefaultAsyncHttp : public ::testing::Test {
+ protected:
+  olp::client::OlpClientSettings client_settings_;
+  olp::client::OlpClient client_;
+};
+
+TEST_F(OlpClientDefaultAsyncHttp, GetGoogleWebsite) {
+  client_.SetBaseUrl("https://www.google.com");
+
+  client_settings_.network_request_handler = olp::client::
```

Comment:
same mistake as before?
Common parts move into SetUp/TearDown

Reply:
Tests refactoring is another huge topic and not in scope of OLPEDGE-717.

Reply:
But this is not an integration test. This is a functional test, resp. online test.
Also not adding the network to as static will crash again. Means separate, wait for crash, fix crash is the workflow?

Reply:
How moving client settings initialization to SetUp will prevent network from being static?  

Yes, wait for crash -> fix crash is the workflow.

Reply:
Two different topics:
1. Network static as in other online tests to avoid crash. Why not also here?
2. As you create a new fixture, move common parts, e.g. client_.SetSettings(client_settings_) to SetUp().

And btw. why is client_settings_ a member?

Reply:
Good question, idk. But it didn't fail once so far. 
I don't create new fixture is was there before. 
Someboby made client_settings_ a member long time ago.

OLPEDGE-717 has a clear scope. I also don't like what this test look like but fixing tests that are not even failing is clearly not in that scope. You are welcome to create another ticket to refactor the tests.

Reply:
Seems tickets scope is very subjective, especially when I see changes here that are also not part of the scope.